Missing rsc-html-stream in package.json of plugin-rsc/samples/ssg

### Related plugins

- [ ] [plugin-react](https://github.com/vitejs/vite-plugin-react/tree/main/packages/plugin-react)

- [ ] [plugin-react-swc](https://github.com/vitejs/vite-plugin-react/tree/main/packages/plugin-react-swc)

- [ ] [plugin-react-oxc](https://github.com/vitejs/vite-plugin-react/tree/main/packages/plugin-react-oxc)

- [x] [plugin-rsc](https://github.com/vitejs/vite-plugin-react/tree/main/packages/plugin-rsc)


### Describe the bug

Files of SSG example in plugin-rsc imports `rsc-html-stream`:

https://github.com/vitejs/vite-plugin-react/blob/954b3610455b72c23caca6964a469af3888e30a4/packages/plugin-rsc/examples/ssg/src/framework/entry.browser.tsx#L7C1-L7C51

But `rsc-html-stream` is not included in the `package.json`:

https://github.com/vitejs/vite-plugin-react/blob/954b3610455b72c23caca6964a469af3888e30a4/packages/plugin-rsc/examples/ssg/package.json#L12-L22




### Reproduction

https://github.com/vitejs/vite-plugin-react/blob/954b3610455b72c23caca6964a469af3888e30a4/packages/plugin-rsc/examples/ssg/

### Steps to reproduce

Run `npm install` followed by `npm run dev`

### System Info

```shell
Binaries:
    Node: 22.16.0
    npm: 10.9.2
  npmPackages:
    @vitejs/plugin-react: ^5.0.2 => 5.0.2
    @vitejs/plugin-rsc: ^0.4.30 => 0.4.30
```

### Used Package Manager

npm

### Logs

```
npm run dev
Debugger attached.

> @vitejs/plugin-rsc-examples-ssg@0.0.0 dev
> vite

Debugger attached.
(!) Failed to run dependency scan. Skipping dependency pre-bundling. Error: The following dependencies are imported but could not be resolved:

  rsc-html-stream/client (imported by .../src/framework/entry.browser.tsx)

Are they installed?
    at file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:11130:33
    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
    at async file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:34823:15
Port 5173 is in use, trying another one...

  VITE v7.1.5  ready in 735 ms

  ➜  Local:   http://localhost:5174/
  ➜  Network: use --host to expose
  ➜  press h + enter to show help
[vite] connected.
[vite] connected.
Error: Cannot find module 'rsc-html-stream/server' imported from '.../src/framework/entry.ssr.tsx'
    at fetchModule (file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:34690:34)
    at RunnableDevEnvironment.fetchModule (file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:35529:10)
    at fetchModule (file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:35498:16)
    at handleInvoke (file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:28797:25)
    at EventEmitter.listenerForInvokeHandler (file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:28858:19)
    at EventEmitter.emit (node:events:518:28)
    at Object.send (file:///.../node_modules/vite/dist/node/chunks/dep-M_KD0XSK.js:18804:39)
    at Object.invoke (file:///.../node_modules/vite/dist/node/module-runner.js:535:31)
    at Object.invoke (file:///.../node_modules/vite/dist/node/module-runner.js:584:31)
    at ModuleRunner.getModuleInformation (file:///.../node_modules/vite/dist/node/module-runner.js:1059:28) {
  code: 'ERR_MODULE_NOT_FOUND',
  runnerError: Error: RunnerError
      at reviveInvokeError (file:///.../node_modules/vite/dist/node/module-runner.js:476:64)
      at Object.invoke (file:///.../node_modules/vite/dist/node/module-runner.js:549:11)
      at async ModuleRunner.getModuleInformation (file:///.../node_modules/vite/dist/node/module-runner.js:1059:7)
      at async ESModulesEvaluator.runInlinedModule (file:///.../node_modules/vite/dist/node/module-runner.js:910:3)
      at async ModuleRunner.directRequest (file:///.../node_modules/vite/dist/node/module-runner.js:1119:59)
      at async ModuleRunner.cachedRequest (file:///.../node_modules/vite/dist/node/module-runner.js:1037:73)
      at async ModuleRunner.import (file:///.../node_modules/vite/dist/node/module-runner.js:997:10)
      at async handler (.../src/framework/entry.rsc.tsx:27:15)
}
```

### Validations

- [x] Follow our [Code of Conduct](https://github.com/vitejs/vite-plugin-react/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guidelines](https://github.com/vitejs/vite-plugin-react/blob/main/CONTRIBUTING.md).
- [x] Read the [docs](https://vite.dev/guide).
- [x] Check that there isn't [already an issue](https://github.com/vitejs/vite-plugin-react/issues) that reports the same bug to avoid creating a duplicate.
- [x] Make sure this is a Vite issue and not a framework-specific issue.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitejs/vite-plugin-react/discussions) or join our [Discord Chat Server](https://chat.vite.dev/).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/packages/plugin-rsc/e2e/isolated.test.ts
+++ b/packages/plugin-rsc/e2e/isolated.test.ts
@@ -4,6 +4,7 @@ import { defineStarterTest } from './starter'
 import path from 'node:path'
 import os from 'node:os'
 import * as vite from 'vite'
+import { waitForHydration } from './helper'
 
 test.describe(() => {
   // use RUNNER_TEMP on Github Actions
@@ -52,5 +53,36 @@ test.describe('vite 6', () => {
   test.describe('build', () => {
     const f = useFixture({ root: tmpRoot, mode: 'build' })
     defineStarterTest(f)
+  })
+})
+
+test.describe('examples/ssg', () => {
+  const tmpRoot = path.join(
+    process.env['RUNNER_TEMP'] || os.tmpdir(),
+    'test-vite-rsc-ssg',
+  )
+  test.beforeAll(async () => {
+    await setupIsolatedFixture({
+      src: 'examples/ssg',
+      dest: tmpRoot,
+    })
+  })
+
+  test.describe('dev', () => {
+    const f = useFixture({ root: tmpRoot, mode: 'dev' })
+
+    test('basic', async ({ page }) => {
+      await page.goto(f.url())
+      await waitForHydration(page)
+    })
+  })
+
+  test.describe('build', () => {
+    const f = useFixture({ root: tmpRoot, mode: 'build' })
+
+    test('basic', async ({ page }) => {
+      await page.goto(f.url())
+      await waitForHydration(page)
+    })
   })
 })

--- a/packages/plugin-rsc/examples/ssg/package.json
+++ b/packages/plugin-rsc/examples/ssg/package.json
@@ -19,6 +19,7 @@
     "@types/react-dom": "^19.1.9",
     "@vitejs/plugin-react": "latest",
     "@vitejs/plugin-rsc": "latest",
-    "rsc-html-stream": "^0.0.7"
+    "rsc-html-stream": "^0.0.7",
+    "vite": "^7.1.5"
   }
 }

--- a/packages/plugin-rsc/examples/ssg/package.json
+++ b/packages/plugin-rsc/examples/ssg/package.json
@@ -18,6 +18,7 @@
     "@types/react": "^19.1.12",
     "@types/react-dom": "^19.1.9",
     "@vitejs/plugin-react": "latest",
-    "@vitejs/plugin-rsc": "latest"
+    "@vitejs/plugin-rsc": "latest",
+    "rsc-html-stream": "^0.0.7"
   }
 }

